BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
```

Comment:
It might be a good idea to be more specific and rename Rates to ExchangeRates, this is just in case we ever expose this in the .Contracts as a nuget package and consume it by other apps. This is so that we don't confuse exchange Rates with hotel pricing Rates in the future for example.

Reply:
Also using a struct here is quite confusing, structs are value types and Rates is a List<Rate> which is a reference type.
Also structs don't support inheritance, so if we ever want to extend these by inheritance, we will not be able to do this. We use classes much more frequently https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/choosing-between-class-and-struct

Reply:
We can actually use an IEnumerable here instead, we only read and iterate over it so using a List for performance reasons shouldn't be the deciding factor.